src/scenarios/shared/runner.py

```diff
@@ -198,11 +206,21 @@ def parseargs(self):
         if args.scenarioname:
             self.scenarioname = args.scenarioname
 
+        if self.testtype in [const.STARTUP, const.INNERLOOP, const.INNERLOOPMSBUILD, const.DOTNETWATCH, const.SDK, const.CROSSGEN, const.CROSSGEN2] and (args.affinity or os.environ.get('PERFLAB_DATA_AFFINITY')): # Set affinity if doing a Startup based test
```

Comment:
Would it be better to use getattr to check if affinity exists instead?

Reply:
Attempted to make a good solution using getattr and couldn't come up with one that worked with both affinity passing options.